incapsula/client_subaccount.go

```diff
@@ -101,22 +103,62 @@ func (c *Client) AddSubAccount(subAccountPayload *SubAccountPayload) (*SubAccoun
 	return &subAccountAddResponse, nil
 }
 
-// ListSubAccounts gets the Incapsula list of SubAccounts
-func (c *Client) ListSubAccounts(accountID int) (*SubAccountListResponse, error) {
+// GetSubAccount gets the Incapsula list of SubAccounts
+func (c *Client) GetSubAccount(parentAccountID int, subAccountID int) (*SubAccount, error) {
 
-	log.Printf("[INFO] Getting Incapsula subaccounts for: %d)\n", accountID)
+	log.Printf("[INFO] Reading Incapsula subaccounts for id: %d)", subAccountID)
+	log.Printf("[INFO] Listing Incapsula subaccounts for: %d (if zero, will be determined by API key))", parentAccountID)
 
+	var count = 0
+	var subAccounts, error = c.sendListSubAccountsRequest(parentAccountID, count)
+	log.Printf("[INFO] read subAccouts list size in response: %d", len(subAccounts))
+	if error != nil {
+		return nil, error
+	}
+	for _, subAccount := range subAccounts {
+		if subAccount.SubAccountID == subAccountID {
+			log.Printf("[INFO] found subaccount : %v\n", subAccount)
+			return &subAccount, nil
+		}
+		count += 1
+	}
+	// Pagination (default page size 50)
+	for len(subAccounts) == PAGE_SIZE {
+		log.Printf("[DEBUG] didn't found subaccount %d, continue fetching for page: %d", subAccountID, count)
```

Comment:
Move this code to separate method  (searchSubAccountOnPage):
 
	log.Printf("[DEBUG] didn't found subaccount %d, continue fetching for page: %d", subAccountID, count)
		subAccounts, error = c.sendListSubAccountsRequest(parentAccountID, count)
		if error != nil {
			return nil, error
		}
		for _, subAccount := range subAccounts {
			if subAccount.SubAccountID == subAccountID {
				log.Printf("[INFO] found subaccount : %v\n", subAccount)
				return &subAccount, nil
			}
		}